pkg/cmd/repo/deploy-key/delete/delete.go

```diff
@@ -0,0 +1,121 @@
+package delete
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/cli/cli/v2/internal/config"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/cli/cli/v2/pkg/prompt"
+	"github.com/spf13/cobra"
+)
+
+type DeleteOptions struct {
+	IO         *iostreams.IOStreams
+	Config     func() (config.Config, error)
+	HTTPClient func() (*http.Client, error)
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	ID          string
+	SkipConfirm bool
+}
+
+func NewCmdDelete(f *cmdutil.Factory, runF func(*DeleteOptions) error) *cobra.Command {
+	opts := &DeleteOptions{
+		HTTPClient: f.HttpClient,
+		Config:     f.Config,
+		IO:         f.IOStreams,
+		BaseRepo:   f.BaseRepo,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "delete [<deploy-key-id>]",
```

Comment:
The argument isn't optional, so square brackets should go away:
```suggestion
		Use:   "delete <key-id>",
```